bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/experiencefragment/CommerceExperienceFragmentImpl.java

```diff
@@ -95,9 +101,30 @@
 
     private Resource xfResource;
     private String name;
+    private AbstractCategoryRetriever categoryRetriever;
```

Comment:
Normally the retrievers should be exposed via getters, so they can be accessed when applying Sling model delegation. But I don't think it's needed here, since the retrievers are only required for identifier lookup and I don't see a use case why the query would be extended.